impl/src/main/java/uk/ac/manchester/cs/owl/owlapi/OWLLiteralImplInteger.java

```diff
@@ -167,9 +167,14 @@ public boolean equals(Object obj) {
             return literal == other.literal;
         }
         if (obj instanceof OWLLiteral) {
-            return ((OWLLiteral) obj).isInteger()
-                && ((OWLLiteral) obj).getLiteral().charAt(0) != '0'
-                && literal == ((OWLLiteral) obj).parseInteger();
+            OWLLiteral other = (OWLLiteral) obj;
+            if (other.isInteger() && other.getLiteral().charAt(0) != '0') {
+                try {
+                    return literal == other.parseInteger();
+                } catch (NumberFormatException e) {
+                    return getLiteral().equals(other.getLiteral());
```

Comment:
I'd prefer to not have to rely on exceptions for the control flow, but to do so requires changes to keep track of which integer literals were created with a value within Java integer range, which comes with its own problems.

compareTo() has the same problem, I'll extend the patch set to fix both.